packages/drivers/odsp-driver/src/odspSnapshotParser.ts

```diff
@@ -38,10 +38,10 @@ function buildHierarchy(flatTree: IOdspSnapshotCommit): ISnapshotTree {
 				unreferenced: entry.unreferenced,
 				groupId: entry.groupId,
 			};
-			node.trees[decodeURIComponent(entryPathBase)] = newTree;
+			node.trees[entryPathBase] = newTree;
```

Comment:
Will removing this result in compat issues? For instance, when this recently broke, the scope was limited because in a lot of cases, snapshots were corectly decoded. With this gone, if there are snapshots out there with encoded path (because of the bug we fixed), they will now cause failures.

Reply:
Yes, but previously encoded(id) == id, so that is why we never faced issues.

Reply:
I meant in the time period where the bug existed where data store Id could be "[".

Reply:
Because the parser for binary snapshot which is used today does not decode it and we didn't get this issue. 
Also, this is for json snapshot which spo is close to retiring.

Reply:
Yes, but didn't we decide that if there would be a fix, it would be in the runtime?

Reply:
we were always encoding on write, and never decoding on read from storage (as we are not using JSON path for boot - the only path that decodes). So, if encoding would change ID, it would not work. And thus, not encoding is not going to break anything, as nobody decoding on the other end.